earth_enterprise/src/fusion/portableglobe/gerewritedbroot.cpp

```diff
@@ -219,21 +228,27 @@ void ReplaceReferencedKml(const std::string &new_kml_base_url,
       if (string_id_or_value->has_value()) {
         // the url is stored directly in the kml_url field
         std::string kml_url = string_id_or_value->value();
-        fout << kml_url << " " << kml_file_name << std::endl;
+        if ( preserve_kml_filenames ) {
+            kml_file_name = kml_file_name.substr(kml_file_name.find_last_of("/") + 1);
```

Comment:
There's a bug here. This should be `kml_file_name = kml_url.substr(kml_url.find_last_of("/") + 1);`. Otherwise, preserve_kml_filenames doesn't do anything.